### ISSUE TITLE
Show the total time for the entire playlist?

### ISSUE THREAD

**Issue by Kayizoku** [mustaqimM/mpv-scripts#5]
I would like to press a hotkey and then it calculates the whole playlist and shows the time it would take to watch the playlist in the same manner as before. It recalculates when playback is sped up. Of course, this would be with a different hotkey. Any thoughts about that?

**Comment by mustaqimM** [comment]
@Kayizoku I don't think that there's an exposed function that the mpv api provides to do this, but one way is using other programmes. Could be something like this with `ffprobe`:

```zsh
fd -e mkv -x ffprobe -i {} -show_entries format=duration -v quiet -of csv="p=0" | awk '{total = total + $1}END{print total}'
```

**Comment by Kayizoku** [comment]
> @Kayizoku I don't think that there's an exposed function that the mpv api provides to do this, but one way is using other programmes. Could be something like this with `ffprobe`:
> 
> ```shell
> fd -e mkv -x ffprobe -i {} -show_entries format=duration -v quiet -of csv="p=0" | awk '{total = total + $1}END{print total}'
> ```

Thank you, could you show me a demonstration and how to use it as a script or on a hotkey? I would greatly appreciate the help :)

**Comment by mustaqimM** [comment]
The snippet accounts for all files in a folder. So it's not exactly what you wanted since you'd need to keep track of the current position in the playlist which is provided by `mp.get_property("playlist")`; the current file will have an index of 1. What's needed is get and add all playback time after index 1 to the end of the list of files that it returns. Sorry, I don't have time currently but search to see how to run external processes with lua

**Comment by Kayizoku** [comment]
> The snippet accounts for all files in a folder. So it's not exactly what you wanted since you'd need to keep track of the current position in the playlist which is provided by `mp.get_property("playlist")`; the current file will have an index of 1. What's needed is get and add all playback time after index 1 to the end of the list of files that it returns. Sorry, I don't have time currently but search to see how to run external processes with lua

No worries, I appreciate the help. Would love to see it in the script too. But might try to tinker with it myself. We'll see :)
